hyperspy/misc/holography/reconstruct.py

```diff
@@ -0,0 +1,207 @@
+
+
+def find_sideband_size(holo_shape, sb_position, sb_size_ratio=0.5):
+    """
+    Finds the size of sideband filter
+
+    Parameters
+    ----------
+    holo_shape : array_like
+            Holographic data array
+    sb_position : tuple
+        The sideband position (y, x), referred to the non-shifted FFT.
+    sb_size_ratio : float, optional
+        Size of sideband as a fraction of the distance to central band
+
+    Returns
+    -------
+    sb_size : float
+        Size of sideband filter
+
+    """
+
+    h = np.array((np.asarray(sb_position) - np.asarray([0, 0]),
+                         np.asarray(sb_position) - np.asarray([0, holo_shape[1]]),
+                         np.asarray(sb_position) - np.asarray([holo_shape[0], 0]),
+                         np.asarray(sb_position) - np.asarray(holo_shape))) * sb_size_ratio
+    return np.min(np.linalg.norm(h, axis=1))
+
+
+def reconstruct(holo_data, holo_sampling, sb_size, sb_position, sb_smoothness, output_shape=None,
+                     plotting=False):
+        """Core function for holographic reconstruction.
+
+        Parameters
+        ----------
+        holo_data : array_like
+            Holographic data array
+        holo_sampling : tuple
+            Sampling rate of the hologram in y and x direction.
+        sb_size : float
+            Size of the sideband filter in pixel.
+        sb_position : tuple
+            Sideband position in pixel.
+        sb_smoothness: float
+            Smoothness of the aperture in pixel.
+        output_shape: tuple, optional
+            New output shape.
+        plotting : boolean
+            Shows details of the reconstruction (i.e. SB selection).
+
+        Returns
+        -------
+            wav : nparray
+                Reconstructed electron wave
+
+        """
+
+        holo_size = holo_data.shape
+        f_sampling = np.divide(1, [a * b for a, b in zip(holo_size, holo_sampling)])
+
+        fft_exp = fft2(holo_data) / np.prod(holo_size)
+
+        f_freq = freq_array(holo_data.shape, holo_sampling)
+
+        sb_size *= np.mean(f_sampling)
+        sb_smoothness *= np.mean(f_sampling)
+        aperture = aperture_function(f_freq, sb_size, sb_smoothness)
+
+        fft_shifted = np.roll(fft_exp, sb_position[0], axis=0)
+        fft_shifted = np.roll(fft_shifted, sb_position[1], axis=1)
+
+        if plotting:
+            fig, axs = plt.subplots(1, 1, figsize=(4, 4))
+            axs.imshow(np.abs(fftshift(fft_shifted * aperture)), clim=(0, 0.1))
+            axs.scatter(sb_position[1], sb_position[0], s=10, color='red', marker='x')
+            axs.set_xlim(int(holo_size[0]/2) - sb_size/np.mean(f_sampling), int(holo_size[0]/2) +
+                         sb_size/np.mean(f_sampling))
+            axs.set_ylim(int(holo_size[1]/2) - sb_size/np.mean(f_sampling), int(holo_size[1]/2) +
+                         sb_size/np.mean(f_sampling))
+            plt.show()
+
+        fft_aperture = fft_shifted * aperture
```

Comment:
What about moving this line above `if plotting` to enable using it inside the if statement?